packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -53,7 +53,11 @@ export class RemoteChannelContext implements IChannelContext {
      * This loads the GC details from the base snapshot of this context.
      */
     private readonly gcDetailsInInitialSummaryP = new LazyPromise<IGarbageCollectionSummaryDetails>(async () => {
-        if (await this.services.objectStorage.contains(gcBlobKey)) {
+        // back-compat 0.34.0 - Older documents may have the GC blob in the channel context's summary. Much older
+        // documents may not have GC blobs at all.
+        if (this.gcDetailsInInitialSummary !== undefined) {
```

Comment:
I do not think this check is required - there is no back compat concerns between data store runtime and DDSs - they exists in the same layer.